backend/src/main/java/com/capstone/moneytree/handler/RestExceptionHandler.java

```diff
@@ -0,0 +1,35 @@
+package com.capstone.moneytree.handler;
+
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import com.capstone.moneytree.exception.EntityNotFoundException;
+import com.capstone.moneytree.utils.MoneyTreeError;
+
+@Order(Ordered.HIGHEST_PRECEDENCE)
+@ControllerAdvice
+public class RestExceptionHandler extends ResponseEntityExceptionHandler {
+
+   private ResponseEntity<MoneyTreeError> buildResponseEntity(MoneyTreeError apiError) {
+      return new ResponseEntity<>(apiError, apiError.getStatus());
+   }
+
+   //other exception handlers below
+   @ExceptionHandler(EntityNotFoundException.class)
+   protected ResponseEntity<MoneyTreeError> handleEntityNotFound(
```

Comment:
Nice! So if we validate an endpoint for example and see that there are no errors, then we would return a ResponseEntity<Object> like before. However, if we have errors, then we would create a ResponseEntity<MoneyTreeError> from the different handles we have here. Then Spring would automatically understand return the right entity?

Reply:
Yeah so, this handler class will handle all exception we indicate. If we throw an exception while processing a request like user not found exception, spring will go to this handler and see which handler handles that specific exception. Then it will return the response entity with our error object.

So there is no need to have any kind of wrapper or anything like that